### ISSUE TITLE
[.NET6] [Source Compat] Epub app doesn't build(NETSDK1004 error) with NET6 SDK

### ISSUE THREAD

**Issue by jiangzeng01** [dotnet/sdk#15937]
**Application Name:**  Epub
**OS:** Windows 10 RS5
**CPU: X64**
**.NET Build Number:** 6.0.100-preview.2.21117.13

**Verify Scenarios:**
1. Windows 10 RS5 X64 + .NET Core SDK build6.0.100-preview.2.21117.13: Fail
2. Windows 10 RS5 X64 + .NET Core SDK build 6.0.100-preview.1.21106.4: Fail
3. Windows 10 RS5 X64 + .NET Core SDK build 5.0.103: Pass
4. Windows 10 RS5 X64 + .NET Core SDK build 3.1.405: Pass

**Repro steps:**
1. Source Code check at [https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015)
2. goto Epub\JFlepp.Epub.Viewer directory
3. dotnet build /p:Platform=x64

**Expected Result:**
Build succeeded

**Actual Result:**
C:\Program Files\dotnet\sdk\6.0.100-preview.2.21117.13\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): error NETSDK1004: Assets file 'C:\Users\valtunc\Desktop\AppSources\Epub\out\obj\JFlepp.Epub.Viewer_dxyrbeao_wpftmp\project.assets.json' not found. Run a NuGet package restore to generate this file. [C:\Users\valtunc\Desktop\AppSources\Epub\JFlepp.Epub.Viewer\JFlepp.Epub.Viewer_dxyrbeao_wpftmp.csproj]

**Binary Log:**
Check at [https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015)

@dotnet-actwx-bot @dotnet/compat

**Comment by saucecontrol** [comment]
Having the same issue here with 5.0.200-preview.21079.7, from VS 16.9p4.

C:\Program Files\dotnet\sdk\5.0.200-preview.21079.7\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): error NETSDK1004: Assets file 'C:\gitlocal\photosauce\out\obj\VisualTest_i1eolmcr_wpftmp\project.assets.json' not found. Run a NuGet package restore to generate this file. [C:\gitlocal\photosauce\tests\VisualTest\VisualTest_i1eolmcr_wpftmp.csproj]

**Comment by jiangzeng01** [comment]
@marcpopMSFT Could you please take a look at this bug or let us know if we assigned wrong person?

**Comment by marcpopMSFT** [comment]
The change that caused this was reverted but it didn't make it into preview 1: https://github.com/dotnet/sdk/pull/15695.  See https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0-known-issues.md#net-sdk for the workaround.

**Comment by jiangzeng01** [comment]
@marcpopMSFT  Thanks, we will try with Preview 2 branch builds.

**Comment by jiangzeng01** [comment]
@marcpopMSFT It looks like the revert [https://github.com/dotnet/sdk/pull/15695](https://github.com/dotnet/sdk/pull/15695) has merged into master by [https://github.com/dotnet/sdk/pull/15776](https://github.com/dotnet/sdk/pull/15776). 

I just tried with latest master build: 6.0.100-preview.3.21123.4, it still failed with same issue:
```
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21123.4\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): error NETSDK1004: Assets file 'F:\repo\Epub\out\obj\JFlepp.Epub.Viewer_lpvxpazt_wpftmp\project.assets.json' not found. Run a NuGet package restore to generate this file. [F:\repo\Epub\JFlepp.Epub.Viewer\JFlepp.Epub.Viewer_lpvxpazt_wpftmp.csproj]
```

Also I tried the Preview 2 branch build 6.0.100-preview.2.21127.3 which is available today, issue is still happening.

Could you please help to take a look at this?
(The workaround works fine for us)

**Comment by marcpopMSFT** [comment]
@sfoslund See above.  Your revert of IncludePackageReferencesDuringMarkupCompilation I was pretty sure should be in master and the preview 2 branch.  Does he have to do a clean build maybe or something else causing this same failure pattern?

**Comment by sfoslund** [comment]
Yes, I just double checked and the revert has been checked into master and preview 2 for almost a month. @jiangzeng01 is this issue repro-ing from a clean build?

**Comment by jiangzeng01** [comment]
@marcpopMSFT  @sfoslund  I just tried with a clean machine with .NET 6 Preview 2 SDK 6.0.100-preview.2.21152.1 installed, copy app source, tried build and rebuild, the issue is still happening, could you please take a look to see if anything wrong here? Repro steps you can see in [https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1276015)

**Comment by sfoslund** [comment]
I'm able to repro this issue with the latest preview 2 build (6.0.100-preview.2.21153.5). It appears `IncludePackageReferencesDuringMarkupCompilation` is being set in Microsoft.WinFX.targets: https://github.com/dotnet/wpf/commit/42fd57cb6911d2c67ecdd9aa9647e5226ea0f232

cc @ryalanms
